src/basis.jl

```diff
@@ -124,12 +177,14 @@ end
 TODO: needs docs 
 """ 
 function radbasis_params(; 
-         r0 = 2.5, 
-         rcut = 5.0,
-         rin = 0.5 * r0,
-         pcut = 2,
-         pin = 2
-         )
+      rin = nothing, 
```

Comment:
I've changed the default, because I think it should be different for pair and rpi basis. Smooth inner cutoff at `0.5 * r0` for rpi and no inner cutoff for pair. Please correct me if I'm wrong! The defaults are now set at `rpi_basis_params` and `pair_basis_params`.

Reply:
Following `ACE1.utils.jl`, we need to have `RPI_basis` (`rpi_basis`) and `PolyPairBasis` (`pair_basis`) and both need `TransformedPolys` (`rad_basis`) for the radial part. So even though pair basis is effectively a radial basis, we still need all three (that is, can't use `pair_basis` in place of `rad_basis`), because `pair_basis` turn `rad_basis`/`TransformedPolys` into `PolyPairBasis`. 

I was inconsistent with where from/how `rpi_basis` and `pair_basis` get the `rad_basis` - the former currently calls `generate_rpi_radbasis` from `generate_rpi_basis` and the later constructs `radbasis` directly in `generate_pair_basis`. To be consistent, I'll make both of them use `generate_radbasis`  (renamed from what is currently `generate_rpi_radbasis`). Maybe that's exactly what you meant? 

Somewhat more removed - I considered if it's possible or worth not to separate `basis` into `rpi_basis` and `pair_basis`. Then, to have the same behaviour as now, `ace_fit` parameters would have an entry: `fit_ace_params["basis"] = [rpi_basis_params, pair_basis_params]`. The parameters would get transformed into `RPI_basis` and `PolyPairBasis` by the same `generate_basis`, depending on the the parameters (e.g. a `type` field or parameters with `N=1` -> `PolyPairBasis` and `N>1` -> `RPI_basis`), and `IPSuperBasis` would be constructed from`[RPI_basis, PolyPairBasis]` like now. On the upside, it would be straightforward to mix and match what basis get combined into `IPSuperBasis` and add new ones without changing the user-facing behaviour. On the other hand, if `RPI_basis` + `PolyPairBasis` is all that is going to be used in ACE1, then I think the current slightly more explicit/verbose setup is more transparent and less error-prone.

Reply:
I like the idea of putting `pair_basis` into `ace_basis`, but I see no reason to have `pair_rad_params` and `rpi_rad_params`. There should simply be 
* `pair_params` : parameters for pair basis 
* `rad_params` : parameters for radial basis for RPI basis.

What case do you have in mind that this doesn't cover?

Reply:
I think you are suggesting this: 

```
function pair_params(species, maxdeg, r0, rin, pin, rcut, pcut, transform)
function rpi_params(species, N, maxdeg, r0, rad_basis, transform, degree)
function rad_params(rin, pin, rcut, pcut)
```

I am suggesting this: 
```
function pair_params(species, maxdeg, r0, rad_basis, transform)
function rpi_params(species, N, maxdeg, r0, rad_basis, transform, degree)
function rad_params(rin, pin, rcut, pcut)
```
both `rpi_params`/`generate_rpi_basis` and `pair_params`/`generate_pair_basis` would be using the same `rad_params`/`generate_rad_basis`, but would have different default values for `rin`& `pin` in `rad_params`.

Reply:
What you are suggesting makes sense in the context of the ACE1 code, but I thought it would be inconvenient for the user. But in the end you are the user and not I, so if you prefer this structure, then let's leave it like this. 

On the other point - I still like your suggestion to put pair_basis into ace_basis. I think this is worthwhile.

Reply:
Ah, I see what you meant in the original comment, I was thinking about it with ACE1 code in mind. 

I'll incorporate `pair_basis` into `ace_basis`. 

I didn't think `basis` was too generic and I could interpret `ace_basis` to mean both `rpi_basis` only and `rpi_basis` & `pair_basis` (which one is correct in general?). Also, currently there already are `fit_ace` and `ace_params`, so `basis` might be clear enough? What do you think?

Reply:
I'm totally ok with just `basis` especially if it includes both the ACE and the pair basis.